packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -555,7 +568,9 @@ export class GarbageCollector implements IGarbageCollector {
         }
 
         // Whether we are running in test mode. In this mode, unreferenced nodes are immediately deleted.
-        this.testMode = this.mc.config.getBoolean(gcTestModeKey) ?? this.gcOptions.runGCInTestMode === true;
+        this.testMode =
+            !this.sweepV0 && // Disable testMode if we're doing Sweep V0
+            (this.mc.config.getBoolean(gcTestModeKey) ?? this.gcOptions.runGCInTestMode === true);
```

Comment:
It feels like you want to imply Fluid.GarbageCollection.TestOverride.SweepV0TimeoutMs => gcTestModeKey
And other places where you test for presence of this.sweepV0TimeoutMs should be actually testing for presence of test mode.
If so, maybe create a tiny wrapper around this.mc.config.getBoolean(gcTestModeKey) that would return true if the other feature gate is set, but otherwise the rest of the code should use SweepV0TimeoutMs only to overwrite this.sweepTimeoutMs?

Reply:
Yeah, this kind of stuff needs to be better encapsulated.  I opened https://dev.azure.com/fluidframework/internal/_workitems/edit/1361 about this and I think it covers your feedback.